.github/workflows/golden_tests.yml

```
@@ -0,0 +1,34 @@
+          flutter-version: '3.24.0'
+          channel: 'stable'
+
+      - name: Install dependencies
+        run: flutter pub get
+
+      - name: Run golden tests
+        run: flutter test test/goldens/
+
+      - name: Upload failures (if any)
+        if: failure()
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.10)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)